### DISCUSSION TITLE
Cannot upgrade to 2.7.1 on Windows 7: "no QT platform plugin could be initialized."

### DISCUSSION THREAD

**Discussion by unalignedcoder** [keepassxreboot/keepassxc#7941] [Q&A]
I had no problem with the previous version, but after having installed version 2.7.1 I cannot run the program anymore.

It says: "no QT platform plugin could be initialized. Reinstalling the application may fix this problem. Available platform plugins are: windows"

My installation is portable, so at first I thought of simply overwriting the files in the folder from the .zip file.

After getting the above error, I used the .msi legacy file to install. Same result.

**Comment by unalignedcoder** [comment]
So it seems that the only way to update the program is to remove it, reinstall it from scratch, and copy the `config` folder and the `.portable` file back in place. 
A bit annoying. Why simply overwriting the folder doesn't work?

**Comment by droidmonkey** [comment]
> copy the config folder and the .portable file back in place

What? If you are _installing_ the application then you SHOULD NOT be using a portable config.

**Reply by unalignedcoder** [reply]
As I explained, I installed on top of a portable installation to see if the error would go away. 

At first, I simply copied the new files overwriting the old (using the windows.legacy.zip), and got an error.

Then tried the installer as a just in case, and got the error too.

Finally, seeing that none of this worked, I installed brand new, then copied the portable files.

Probably could have created another portable installation from the zip and copied files there _to the same effect._

Bottom line is : **you cannot overwrite a portable installation as a way to upgrade.**  

At least, I couldn't.

Also, I don't see how any of this should cause that plugin platform error.

**Reply by droidmonkey** [reply]
When you drag and drop on top of an existing install you do not _remove_ files that are no longer necessary or which may cause dll load order confusion. It does explain the error because there were old files in the runtime directory that were likely being used that shouldn't have been there.

**Reply by unalignedcoder** [reply]
> When you drag and drop on top of an existing install you do not _remove_ files that are no longer necessary or which may cause dll load order confusion. It does explain the error because there were old files in the runtime directory that were likely being used that shouldn't have been there.

So how is the user of a portable installation advised to proceed with an upgrade?

**Reply by droidmonkey** [reply]
unzip to new directory, copy config files over.
